### PR TITLE
Get rid of the dirs dependency. #1577

### LINKED ISSUE #1577: Get rid of the `dirs` dependency.
**Issue summary**

 - [ ] Remove `utils::get_home_dir`.
 - [ ] ~Merge the default instance for `cli::Config` and `cli::find_default_config`.~
 - [ ] Remove `dirs` dependency.

**Other information and links**

### PR DESCRIPTION
**Pull request by tyshko5**
**Summary of changes**
Changes introduced in this pull request:
- 

**Reference issue to close (if applicable)**

Closes #1577

**Other information and links**

### CODE REVIEW

File: forest/src/cli/config.rs [hunk 1]
```diff
@@ -34,9 +34,15 @@ pub struct Config {
 
 impl Default for Config {
     fn default() -> Self {
+        let dir = ProjectDirs::from("com", "ChainSafe", "Forest").expect("project directories couldn't be found and that FOREST_CONFIG_PATH must be set manually.");
```
Comment (lemmih) [inline forest/src/cli/config.rs:37]: `failed to find project directories, please set FOREST_CONFIG_PATH environment variable manually.`

File: forest/src/cli/config.rs [hunk 2]
```diff
@@ -34,9 +34,15 @@ pub struct Config {
 
 impl Default for Config {
     fn default() -> Self {
+        let dir = ProjectDirs::from("com", "ChainSafe", "Forest").expect("failed to find project directories, please set FOREST_CONFIG_PATH environment variable manually.");
         Self {
             network: Libp2pConfig::default(),
-            data_dir: get_home_dir() + "/.forest",
+            data_dir: dir
```
Comment (LesnyRumcajs) [inline forest/src/cli/config.rs:40]: In general, why don't we want to hold the `data_dir` in a `PathBuf`? Seems more useful and safer to have properly typed paths. Not a blocker for me, just a suggestion.

**Review by LesnyRumcajs — APPROVED** [review]

File: forest/src/cli/config.rs (hunk 2 above)
Comment (lemmih) [reply forest/src/cli/config.rs:40]: Yes, `data_dir` should be a `PathBuf`. The toml parsing still works fine on PathBufs.

**Review by lemmih — CHANGES_REQUESTED** [review]
Change `data_dir` to be a `PathBuf` and then it's all set.

File: forest/src/daemon.rs [hunk 3]
```diff
@@ -39,7 +39,8 @@ pub(super) async fn start(config: Config) {
         option_env!("FOREST_VERSION").unwrap_or(env!("CARGO_PKG_VERSION"))
     );
 
-    let path: PathBuf = [&config.data_dir, "libp2p"].iter().collect();
+    let mut path: PathBuf = config.data_dir.clone();
```
Comment (LesnyRumcajs) [inline forest/src/daemon.rs:42]: Would it be cleaner to have `let path = config.data_dir.join("libp2p")` or am I missing something?

**Review by LesnyRumcajs — CHANGES_REQUESTED** [review]

File: forest/src/cli/config.rs [hunk 4]
```diff
@@ -34,9 +35,10 @@ pub struct Config {
 
 impl Default for Config {
     fn default() -> Self {
+        let dir = ProjectDirs::from("com", "ChainSafe", "Forest").expect("failed to find project directories, please set FOREST_CONFIG_PATH environment variable manually.");
```
Comment (lemmih) [inline forest/src/cli/config.rs:38]: Ideally we would factor this out so we only have a single source of truth for the ProjectDirs. But that can wait until another PR.

**Review by lemmih — APPROVED** [review]

**Review by LesnyRumcajs — APPROVED** [review]

**Pull request merged**